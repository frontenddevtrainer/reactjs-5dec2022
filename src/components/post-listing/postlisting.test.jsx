import { render, screen } from "@testing-library/react";
import PostListing from "./postlisting"
import { BrowserRouter } from "react-router-dom";

describe("Snapshot testing for PostListing", ()=>{
    test("Render PostListing without data", ()=>{

        const { asFragment } = render(<PostListing/>);
        const content = asFragment()
        expect(content).toMatchSnapshot();

    })
    

    test("Render PostListing with data", ()=>{

        const mockdata = [{ userId: 1, id: 1, title: "mock title", body: "mock content" }]

        const { asFragment } = render(<BrowserRouter>
            <PostListing  data={mockdata}/>
        </BrowserRouter>
        
        );
        const content = asFragment()
        expect(content).toMatchSnapshot();

    })
})

describe("UT testing for PostListing", ()=>{
    
})