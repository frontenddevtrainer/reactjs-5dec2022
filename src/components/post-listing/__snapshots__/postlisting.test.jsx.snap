// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing for PostListing Render PostListing with data 1`] = `
<DocumentFragment>
  <div>
    <div
      style="padding: 10px;"
    >
      <h1
        style="font-size: 20px; margin: 0px 0px 10px 0px;"
      >
        mock title
      </h1>
      <div>
        mock content
      </div>
      <div>
        <a
          href="/posts/1/1"
        >
          Read more
        </a>
         
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Snapshot testing for PostListing Render PostListing without data 1`] = `
<DocumentFragment>
  <div>
    No records found!.
  </div>
</DocumentFragment>
`;
